### DISCUSSION TITLE
Nordic BMD-360 GPIO Callback Not Being Called

### DISCUSSION THREAD

**Discussion by Lotharyx** [zephyrproject-rtos/zephyr#69048] [Q&A] [answered]
I'm developing against a Nordic BMD-360 module; for the moment, using the ublox BMD-300 evaluation board.

I've come up with the following code to initialize a GPIO pin that I want to serve as the source of notifications (in this example, pin 16 which is connected to a button on the eval board):

```
#define GPIO0_DEV DEVICE_DT_GET(DT_NODELABEL(gpio0))
#define RTS_DEV GPIO0_DEV
#define RTS_PIN 16
#define RTS_FLAGS (GPIO_INPUT | GPIO_INT_LEVEL_ACTIVE)

void gpio_callback(const struct device * port, struct gpio_callback * cb, gpio_port_pins_t pins) {
	uart_printf("GPIO callback!  Pins = 0x%08x is now %s\r\n", (uint32_t) pins, gpio_pin_get(RTS_DEV, RTS_PIN) ? "high" : "low");
}

void gpio_init(void) {
	int r = gpio_pin_interrupt_configure(RTS_DEV, RTS_PIN, RTS_FLAGS); 
	uart_printf("RTS config result: %d\r\n", r);
	struct gpio_callback callbackstruct = { 0 };
	gpio_init_callback(&callbackstruct, gpio_callback, 1 << RTS_PIN);
	r = gpio_add_callback(RTS_DEV, &callbackstruct);
	uart_printf("add_callback -> %d\r\n", r);
}
```

The code compiles and runs fine; on my serial terminal that receives the uart_printf output, I see:
```
RTS config result: 0
CTS config result: 0
add_callback -> 0 
```
...which tells me no errors are returned by any of the `gpio_` calls.  However, generating input on pin 16, whether by pressing the button or using a jumper wire to connect it to GND or +V, does not cause the callback to be called.

What am I missing?  What do I need to check to find out why the callback isn't being called?  What other information do I need to provide?  My Zephyr SDK version is 0.16.1.

**Comment by nixward** [comment, marked as answer]
You first need to configure the pin as an input. Do that first before setting the call back.

**Reply by Lotharyx** [reply]
Ah yes, I just figured that bit out myself before you chimed in; need to call both `gpio_pin_configure` and `gpio_pin_interrupt_configure` in that order.  Thanks for the response!

Now my code is crashing when the interrupt happens, but that's a story for another day, I think...
